### PR TITLE
[14.0][FIX] account_fiscal_year_closing, migration script for fiscal year multi company rule

### PR DESCRIPTION
**Pull request by GSLabIt**
migration script for fiscal year multi company rule, fix noupdate changes on installed modules

### CODE REVIEW

**Comment by francesco-ooops** [comment]
@matteoopenf since you did migration to v14, does this look good for you?

**Comment by matteoopenf** [comment]
> @matteoopenf since you did migration to v14, does this look good for you?

I think is ok, @GSLabIt  you could fix pre-commit?
After fix I will approve

**Comment by GSLabIt** [comment]
> > @matteoopenf since you did migration to v14, does this look good for you?
> 
> I think is ok, @GSLabIt you could fix pre-commit? After fix I will approve

not a pr issue, it's generalized. See https://github.com/OCA/oca-addons-repo-template/pull/214

**Comment by matteoopenf** [comment]
> > > @matteoopenf since you did migration to v14, does this look good for you?
> > 
> > 
> > I think is ok, @GSLabIt you could fix pre-commit? After fix I will approve
> 
> not a pr issue, it's generalized. See [OCA/oca-addons-repo-template#214](https://github.com/OCA/oca-addons-repo-template/pull/214)

ok I read the conversation we should wait the issue is resolved, when solved tag me again and I will approve, because if the precommit fail, is not possibile merge

**Comment by GSLabIt** [comment]
> > > > @matteoopenf since you did migration to v14, does this look good for you?
> > > 
> > > 
> > > I think is ok, @GSLabIt you could fix pre-commit? After fix I will approve
> > 
> > not a pr issue, it's generalized. See [OCA/oca-addons-repo-template#214](https://github.com/OCA/oca-addons-repo-template/pull/214)
> 
> ok I read the conversation we should wait the issue is resolved, when solved tag me again and I will approve, because if the precommit fail, is not possibile merge 
> 

you could approve anyway imho

**Review by matteoopenf — APPROVED** [review]
LGTM

**Comment by matteoopenf** [comment]
> > > > > @matteoopenf since you did migration to v14, does this look good for you?
> > > > 
> > > > 
> > > > I think is ok, @GSLabIt you could fix pre-commit? After fix I will approve
> > > 
> > > 
> > > not a pr issue, it's generalized. See [OCA/oca-addons-repo-template#214](https://github.com/OCA/oca-addons-repo-template/pull/214)
> > 
> > 
> > ok I read the conversation we should wait the issue is resolved, when solved tag me again and I will approve, because if the precommit fail, is not possibile merge
> 
> you could approve anyway imho

done

**Comment by pedrobaeza** [comment]
If you are in a hurry, you can apply manually the patch in the repo in a different PR, like this one:

https://github.com/OCA/sale-workflow/pull/2748

**Comment by GSLabIt** [comment]
> If you are in a hurry, you can apply manually the patch in the repo in a different PR, like this one:
> 
> [OCA/sale-workflow#2748](https://github.com/OCA/sale-workflow/pull/2748)

Not in a hurry, but thanks for the hint

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml
Comment (sergiocorato) [inline account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: Meaning of this change?

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml
Comment (sergiocorato) [inline account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: Meaning of this change?

**Review by sergiocorato — COMMENTED** [review]
Thanks, I added some comments.

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml
Comment (GSLabIt) [reply account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: > Meaning of this change?

https://github.com/OCA/account-closing/pull/276

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml
Comment (GSLabIt) [reply account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: > Meaning of this change?

https://github.com/OCA/account-closing/pull/276

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml
Comment (sergiocorato) [reply account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: Ok, thanks

File: account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml
Comment (sergiocorato) [reply account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: This is a form, perhaps was intended to fix this row? 
account_fiscal_year_closing/views/account_fiscalyear_closing_views.xml:10 Deprecated "<tree string=..." - [xml-deprecated-tree-attribute]

Comment (GSLabIt) [reply account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: > This is a form, perhaps was intended to fix this row? account_fiscal_year_closing/views/account_fiscalyear_closing_views.xml:10 Deprecated "<tree string=..." - [xml-deprecated-tree-attribute]

sorry wrong fix, now should be ok

Comment (sergiocorato) [reply account_fiscal_year_closing/views/account_fiscalyear_closing_template_views.xml]: Thanks

**Review by sergiocorato — APPROVED** [review]
LGTM

**Comment by sergiocorato** [comment]
I'll merge this PR even if pre-commit is failed, as #276 is blocked.

**Comment by francesco-ooops** [comment]
@sergiocorato can you make a PR to fix dotfiles as done for example in https://github.com/OCA/purchase-workflow/pull/2075 or https://github.com/OCA/l10n-italy/pull/3687 ?

We're working on fixing other repos. Thanks!

**Comment by sergiocorato** [comment]
> I'll merge this PR even if pre-commit is failed, as #276 is blocked.

It's not possible I think, @etobella please put in #276 all the pre-commit fixes.

**Comment by francesco-ooops** [comment]
> @sergiocorato can you make a PR to fix dotfiles as done for example in [OCA/purchase-workflow#2075](https://github.com/OCA/purchase-workflow/pull/2075) or [OCA/l10n-italy#3687](https://github.com/OCA/l10n-italy/pull/3687) ?
> 
> We're working on fixing other repos. Thanks!

@sergiocorato you can just open a new PR with the necessary fixes as we've done in the linked PRs above ;)

**Comment by sergiocorato** [comment]
> > @sergiocorato can you make a PR to fix dotfiles as done for example in [OCA/purchase-workflow#2075](https://github.com/OCA/purchase-workflow/pull/2075) or [OCA/l10n-italy#3687](https://github.com/OCA/l10n-italy/pull/3687) ?
> > We're working on fixing other repos. Thanks!
> 
> @sergiocorato you can just open a new PR with the necessary fixes as we've done in the linked PRs above ;)

#276 already does the copier template update to 1.17.2 (as done in [OCA/purchase-workflow#2075](https://github.com/OCA/purchase-workflow/pull/2075) ), to be mergeable the same PR has to do all the fixes.

**Comment by francesco-ooops** [comment]
@sergiocorato we superseded @etobella 's PRs to help him get the work done, then he merged them. Or maybe there's something I'm not grasping here

**Comment by sergiocorato** [comment]
> @sergiocorato we superseded @etobella 's PRs to help him get the work done, then he merged them. Or maybe there's something I'm not grasping here

PRs are not mergeables if copier template is not updated, so doing the fixes in PR without updating copier template in the same time is un-mergeable.

I'll do a PR to #276 to make it mergeable.

**Comment by francesco-ooops** [comment]
@sergiocorato merge?

**Comment by sergiocorato** [comment]
> @sergiocorato merge?

I remember that 2 positive reviews are required to merge.

**Comment by francesco-ooops** [comment]
@OCA/accounting-maintainers anyone can take a look at this migration script fix? thanks!

**Comment by sergiocorato** [comment]
> @OCA/accounting-maintainers anyone can take a look at this migration script fix? thanks!

Anyone can review this PR.

**Comment by francesco-ooops** [comment]
Then we do have two positive reviews :)

image

**Comment by sergiocorato** [comment]
Sorry, I'll change my glasses :)
/ocabot merge nobump

**Pull request merged**
